Makefile

```diff
@@ -67,4 +70,11 @@ clean:
 	rm -f SDL/*.o
 	cd efficiency/ && make clean
 
-.PHONY: $(ROOUTIL) efficiency
+format:
+	clang-format -i SDL/*.cc SDL/*.h
+
+check:
+	clang-tidy SDL/*.cc SDL/*.h -- --language=c++ -std=c++17 -I. -I$(ROOT_ROOT)/include -I$(CMSSW_BASE)/src -I${ALPAKA_ROOT}/include -I/${BOOST_ROOT}/include -DALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED -I/cvmfs/cms.cern.ch/el8_amd64_gcc11/external/gcc/11.4.1-30ebdc301ebd200f2ae0e3d880258e65/include/c++/11.4.1 -I/cvmfs/cms.cern.ch/el8_amd64_gcc11/external/gcc/11.4.1-30ebdc301ebd200f2ae0e3d880258e65/include/c++/11.4.1/x86_64-redhat-linux-gnu/
```

Comment:
To avoid copy-paste I'd use some common var to be used both in `check` and `check-fix`.
Is it possible to have this rule in `SDL/Makefile` instead?

for the gcc part I used a probably more maintainable  
```
`g++ -E -x c++ - -v < /dev/null 2>&1 | sed -e 's/^ //g' | awk '/starts here:/{p="";s=1}/End of search/{ms=p;s=1}{if(s==0)p=p" -I"$0;s=0}END{print ms}'`
```

Also, our default build includes
`-DT4FromT3 -DCACHE_ALLOC -DDUP_pLS -DDUP_T5 -DDUP_pT5 -DDUP_pT3 -DCrossclean_T5 -DCrossclean_pT3 -DPT_CUT=0.8`
at least some of these change the code visible to the compiler.

Reply:
Yeah, those are very good points. I'll take care of those.